src/main/kotlin/baseball/IOHandler.kt

```diff
@@ -0,0 +1,41 @@
+package baseball
+
+import baseball.data.GameNumDeck
+import baseball.data.InputType
+import camp.nextstep.edu.missionutils.Console
+
+class IOHandler {
+
+    private val validInputAfterGame = listOf("1", "2")
+    private val validInputWhileGame = (1..10).map { (it + ASCII_0_CODE).toChar() }
+
+    fun getInput(inputType: InputType): String {
+        val input = Console.readLine()
+
+        if (checkInputValid(input, inputType).not()) {
+            throw IllegalArgumentException()
+        }
+        return input
+    }
+
+    fun show(msg: String) = print(msg)
+
+    private fun checkInputValid(input: String, inputType: InputType): Boolean {
+        return when (inputType) {
+            InputType.WHILE_GAME -> checkInputValidWhileGame(input)
+            InputType.AFTER_GAME -> checkInputValidAfterGame(input)
+        }
+    }
+
+    private fun checkInputValidWhileGame(input: String): Boolean {
+        return input.filter {
+            it in validInputWhileGame
+        }.toSet().size == GameNumDeck.DIGIT
```

Comment:
filter 마지막에 . 2개와 '==' 연산자 까지 같은 라인으로 구성되어 있는데, 가독성을 위해 아래와 같이 수정하는 것은 어떨까요?  **변수 네이밍은 무시해주세요 ㅜㅜ**
```suggestion
        val inputUniqueLength = input.filter {
            it in validInputWhileGame
        }.toSet().size
        
        return inputUniqueLength == GameNumDeck.DIGIT
```

Reply:
더 가독성이 좋은 것 같습니다! 아이디어 감사합니다!

이 함수를 구현할 때, 입력한 숫자가 3개인지 확인하는 것과 3개의 값이 모두 다른지 확인하는 것을 함께 처리하고 싶어서 set을 이용했었는데요.
7644 와 같은 값을 넣으면 예외가 발생하는 잘못된 함수입니다.

아래 세가지 조건을 최대한 압축할 수 있는 좋은 아이디어가 나중에 떠오르시면 알려주세요😃
1. 입력한 값의 길이는 3인가
2. 입력한 String에 있는 각 값은, 1~9에 속하는 가
3. 입력한 값이 모두 서로 다른가

Reply:
아~ 그렇게 생각을 못해봤는데, 7644 같은 케이스가 예외로 발생하네요.
`if (input.length != 3) return false` 이 코드를 inputUniqueLength 변수 선언 전에 넣어준다면 가능하지 않을까 생각됩니다!

기존 코드에서 본다면 많이 길어지긴 하지만 아래와 같이 할 수 있을 것 같아요!
```kotlin
return if (input.length != 3) false else input.filter {
    it in validInputWhileGame
}.toSet().size == GameNumDeck.DIGIT
```